modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/resolver/ResolverUtils.java

```diff
@@ -139,36 +140,38 @@ public static Object resolveCursor(Object value) {
                                                                   MuleContext muleContext) {
 
     try {
+      ValueResolver resolver;
       if (stackedTypesModelProperty.isPresent()) {
-        return stackedTypesModelProperty.get().getValueResolverFactory().getExpressionBasedValueResolver(expression,
-                                                                                                         getType(type));
+        resolver = stackedTypesModelProperty.get().getValueResolverFactory().getExpressionBasedValueResolver(expression,
+                                                                                                             getType(type));
         //TODO MULE-13518: Add support for stacked value resolvers for @Parameter inside pojos // The following "IFs" should be removed once implemented
       } else if (isTypedValue.getAsBoolean()) {
         ExpressionTypedValueValueResolver<Object> valueResolver =
             new ExpressionTypedValueValueResolver<>(expression, getType(type));
         valueResolver.setTransformationService(muleContext.getTransformationService());
         valueResolver.setExtendedExpressionManager(muleContext.getExpressionManager());
-        return valueResolver;
+        resolver = valueResolver;
       } else if (isParameterResolver.getAsBoolean()) {
         ExpressionBasedParameterResolverValueResolver<Object> valueResolver =
             new ExpressionBasedParameterResolverValueResolver<>(expression, getType(type), type);
         valueResolver.setTransformationService(muleContext.getTransformationService());
         valueResolver.setExtendedExpressionManager(muleContext.getExpressionManager());
-        return valueResolver;
+        resolver = valueResolver;
       } else if (muleContext.getExpressionManager().isExpression(expression)) {
         TypeSafeExpressionValueResolver<Object> valueResolver =
             new TypeSafeExpressionValueResolver<>(expression, getType(type), type);
         valueResolver.setTransformationService(muleContext.getTransformationService());
         valueResolver.setExtendedExpressionManager(muleContext.getExpressionManager());
-        valueResolver.initialise();
-        return valueResolver;
+        resolver = valueResolver;
       } else {
         TypeSafeValueResolverWrapper typeSafeValueResolverWrapper =
             new TypeSafeValueResolverWrapper<>(new StaticValueResolver<>(expression), getType(type));
         typeSafeValueResolverWrapper.setTransformationService(muleContext.getTransformationService());
-        typeSafeValueResolverWrapper.initialise();
-        return typeSafeValueResolverWrapper;
+        resolver = typeSafeValueResolverWrapper;
       }
+
+      initialiseIfNeeded(resolver, muleContext);
```

Comment:
again, pretty sure this resolver will be initialised again at some point. Additionally, I get the impression you're calling this full initialisation method when all you really want is to perform DI.

Reply:
some were being initialised and some did not. This is normalization